### PR TITLE
BuildResidentialHPXML measure: Fixes for cathedral ceilings

### PR DESCRIPTION
**Pull request by aspeake**
## Pull Request Description

Fixes the following bugs in the BuildResidentialHPXML measure for buildings with cathedral ceilings:
1. Outside boundary condition of gable walls on a shared facade is `outdoors` instead of `adiabatic` (SFA only)
2. No roof surface ids are set as attached to the attic in the xml (required when running with HES)

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [ ] ~~Schematron validator (`EPvalidator.xml`) has been updated~~
- [x] Sample files have been added/updated (via `tasks.rb`)
- [ ] ~~Unit tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests`)~~
- [ ] ~~Documentation has been updated~~
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] Checked the code coverage report on CI
- [x] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: BuildResidentialHPXML/resources/geometry.rb [hunk 1]
```diff
@@ -1816,6 +1816,22 @@ def self.create_single_family_attached(runner:,
       end
     end
 
+    # Adiabatic gable walls for cathedral ceilings
+    if attic_type == HPXML::AtticTypeConditioned
+      attic_space.surfaces.each do |surface|
```
Comment (shorowit) [inline BuildResidentialHPXML/resources/geometry.rb:1821]: It looks like you simply copied code from above? Please don't duplicate code ([DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)), either refactor the code or use a method. In this case, it seems like you can remove lines 1804-1816 and then just update your code here to be:
```suggestion
    # Adiabatic gable walls for cathedral ceilings
    if [HPXML::AtticTypeVented, HPXML::AtticTypeUnvented, HPXML::AtticTypeConditioned].include?(attic_type)
      attic_space.surfaces.each do |surface|
```

File: Changelog.md [hunk 2]
```diff
@@ -38,6 +38,7 @@ __Bugfixes__
 - Adds more decimal places in output files as needed for simulations with shorter timesteps and/or abbreviated run periods.
 - Timeseries output fixes: some outputs off by 1 hour; possible negative combi boiler values.
 - Fixes range hood ventilation interaction with infiltration to take into account the location of the cooking range.
+- Fixes incorrect outside boundary condition for shared gable walls of cathedral ceilings, now set to adiabatic.
```
Comment (shorowit) [inline Changelog.md:41]: If this fixed a bug, why isn't there a change to the gable walls defined in the base-bldgtype-single-family-attached.xml sample file?

File: workflow/sample_files/base-atticroof-cathedral.xml [hunk 3]
```diff
@@ -81,6 +81,7 @@
             <AtticType>
               <CathedralCeiling/>
             </AtticType>
+            <AttachedToRoof idref='Roof1'/>
```
Comment (shorowit) [inline workflow/sample_files/base-atticroof-cathedral.xml:84]: 👍

File: Changelog.md (hunk 2 above)
Comment (aspeake) [reply Changelog.md:41]: That file does not have cathedral ceilings

Comment (shorowit) [reply Changelog.md:41]: Right. Good point. Do you mind creating a new base-bldgtype-single-family-attached-atticroof-cathedral.xml based on base-bldgtype-single-family-attached.xml? Seems like a good test file to have.

**Review by shorowit — APPROVED** [review]
👍 Looks good to me. Thanks @aspeake!

**Pull request merged**